### ISSUE TITLE
[question] Pytorch XLA 1.8 TPU v2-8 code freezes on clip_grad_norm_ 

### ISSUE THREAD

**Issue by nupurkmr9** [pytorch/xla#2847]
I am trying to run stylegan2-pytorch-diffaugment (https://github.com/mit-han-lab/data-efficient-gans) code on TPU. I am using Pytorch-xla-1.8 env and v2-8 TPU. For gradient clipping I am using the following: 
```
from torch_xla._patched_functions import *  
xm.reduce_gradients(phase.opt)
xm.master_print("grad norm clipping")
clip_grad_norm_(phase.module.parameters() , max_norm = 1e5 , norm_type = np.inf)
```

clip_grad_norm takes forever to run. 
Can anyone help me what is going wrong?

**Comment by JackCaoG** [comment]
Hi @nupurkmr9 Thanks for reporting! Any chance you can provide us a short repo that we can run and test?

**Comment by tchaton** [comment]
```
from omegaconf import DictConfig

import torch_xla.core.xla_model as xm
import torch_xla.distributed.xla_multiprocessing as xmp

def fn(process_idx: int):
    d = DictConfig({"1": "1"})
    xm.save(d, "test.pt")

xmp.spawn(fn, nprocs=1)
```

**Comment by tchaton** [comment]
Tagging @omry.

**Comment by nupurkmr9** [comment]
Hi, thanks for the quick response.
I have put the code to replicate this at [https://github.com/nupurkmr9/data-efficient-gans](https://github.com/nupurkmr9/data-efficient-gans)
I have also put a sample data folder of 100 images at:
https://drive.google.com/file/d/1OBR7UXkW2BhPgSNyciK7VFUazUbb5enq/view?usp=sharing

The commands required for replicating the issue is:
cd DiffAugment-stylegan2-pytorch/
copy image folder inside DiffAugment-stylegan2-pytorch
`python train.py --mirror 1 --kimg 5000 --gpus 8 --cfg paper256 --outdir . --data sample_folder/ --fp32 1`

Let me know if anything else is required.

**Comment by JackCaoG** [comment]
@nupurkmr9 Looking through you `train.py` briefly, I didn't see you import `torch_xla`, nor did you use https://github.com/pytorch/xla/blob/ba015ce9a1a5430fcceaf9c21b07d9f210e58e96/test/test_train_mp_imagenet.py#L179 to set the device to be a xla_device. How do you make your code running with the tpu? 

For a reference, you can check our https://github.com/pytorch/xla/blob/master/test/test_train_mp_mnist.py as a minimal example of running your code using pt/xla.

**Comment by nupurkmr9** [comment]
training/training_loop.py is the main script for training loop over dataset. device initialization and torch_xla is imported in that file.

**Comment by JackCaoG** [comment]
Sorry for the delay, I will try to repo this when I am a bit more free,

**Comment by nupurkmr9** [comment]
Sure, Thanks!
I found that whenever I did a for loop like:
 `for p in module.parameters():
              #do something`
the code slows down.
